pkgs/by-name/co/code-cursor/package.nix

```diff
@@ -5,93 +5,143 @@
   appimageTools,
   makeWrapper,
   writeScript,
+  undmg,
 }:
 let
   pname = "cursor";
-  version = "0.44.11";
+  version = "0.45.8";
+
+  inherit (stdenvNoCC) hostPlatform;
 
   sources = {
     x86_64-linux = fetchurl {
-      url = "https://download.todesktop.com/230313mzl4w4u92/cursor-0.44.11-build-250103fqxdt5u9z-x86_64.AppImage";
-      hash = "sha256-eOZuofnpED9F6wic0S9m933Tb7Gq7cb/v0kRDltvFVg=";
+      url = "https://download.todesktop.com/230313mzl4w4u92/cursor-0.45.8-build-250201b44xw1x2k-x86_64.AppImage";
+      hash = "sha256-H+9cisa1LWJleqzwaB0WIzJpioYZyfLghelcZthCOvg=";
     };
     aarch64-linux = fetchurl {
-      url = "https://download.todesktop.com/230313mzl4w4u92/cursor-0.44.11-build-250103fqxdt5u9z-arm64.AppImage";
-      hash = "sha256-mxq7tQJfDccE0QsZDZbaFUKO0Xc141N00ntX3oEYRcc=";
+      url = "https://download.todesktop.com/230313mzl4w4u92/cursor-0.45.8-build-250201b44xw1x2k-arm64.AppImage";
+      hash = "sha256-GgPt9SvuCA9Hxm7nxm7mz0AvPKaLWCkYXO225taXnLA=";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://download.todesktop.com/230313mzl4w4u92/Cursor%200.45.8%20-%20Build%20250201b44xw1x2k-x64.dmg";
+      hash = "sha256-UqwzgxBSZR0itCknKzBClEX3w9aFKFhGIiVUQNYDVEM=";
+    };
+    aarch64-darwin = fetchurl {
+      url = "https://download.todesktop.com/230313mzl4w4u92/Cursor%200.45.8%20-%20Build%20250201b44xw1x2k-arm64.dmg";
+      hash = "sha256-AUW19xJFsrDGSUNE/bwkC2aN2QyaS+JKCjuxx//kbiI=";
     };
   };
 
-  supportedPlatforms = [
-    "x86_64-linux"
-    "aarch64-linux"
-  ];
-
-  src = sources.${stdenvNoCC.hostPlatform.system};
+  source = sources.${hostPlatform.system};
 
+  # Linux -- build from AppImage
   appimageContents = appimageTools.extractType2 {
-    inherit version pname src;
+    inherit version pname;
+    src = source;
   };
 
-  wrappedAppImage = appimageTools.wrapType2 { inherit version pname src; };
+  wrappedAppimage = appimageTools.wrapType2 {
+    inherit version pname;
+    src = source;
+  };
 
-  appimageInstall = ''
```

Comment:
`installPhase` was written as a function in the let block with the idea we would have a linux install phase and a darwin install phase; it proved better to unify the installPhase `mkDerivation` with conditional content. Ignore the pink removals and read the green additions from here to the end of the install phase.